static/js/message_edit.js

```diff
@@ -305,6 +306,19 @@ export function end_if_focused_on_message_row_edit() {
     }
 }
 
+export function update_inline_topic_edit_ui() {
+    // This function is called when "realm_move_messages_within_stream_limit_seconds"
+    // setting is changed.
+
+    if (!settings_data.user_can_move_messages_to_another_topic()) {
+        // There is no need to rerender if user is already not
+        // allowed to edit topics as per "edit_topic_policy".
+        return;
+    }
```

Comment:
Hmm, what happens if the user previously had that permission? I'll drop this hunk for today's merge, we can do a refinement in a follow-up. (I could imagine checking if we had the permission either before or after).

Reply:
So, this condition only checked whether the user had permission to edit topic as per role-based `edit_topic_policy` setting and we call this function only on updating the time-limit setting and not when updating `edit_topic_policy`.